.github/workflows/check.yaml

```
@@ -0,0 +1,33 @@
+name: Analyze code and run tests
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: ⬇️ Checkout repository
+        uses: actions/checkout@v3
+
+      - name: ⚙️ Setup Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          flutter-version: "3.16.3"
+          channel: 'stable'
+          cache: true
+
+      - name: ⚙️ Install dependencies
+        run: flutter pub get
+
+      - name: 📝 Verify formatting
+        run: dart format --output=none --set-exit-if-changed .
+
+      - name: 🧐 Analyze project source
+        run: flutter analyze --fatal-infos
+
+      - name: 🤖 Run tests
+        run: flutter test
```

It will give us formatting prepared for GitHub Actions ;)

```suggestion
        run: dart test -r github
```

---

https://github.com/nomtek/mistralai_client_dart/pull/14/commits/111b38b0e14b62345ed6141db6619876ed8e8527